project/src/main/webapp/document.jsp

```diff
@@ -181,6 +185,18 @@
         xhttp.send(formData);
         return false;
       }
+
+      //Downloads current doument
+      function download() {
+        var text = firepad.getText();
```

Comment:
do we need to check whether firepad is initialized here?

Reply:
The firepad is initialized onload so it must be initialized before we can even interact with the button that calls this function. If an error is encountered then it will be dealt with before getting to this